### ISSUE TITLE
Failed to convert int16 array

### ISSUE THREAD

**Issue by garaemon** [jsk-ros-pkg/geneus#4]
I have error in calling `(send *ri* :state :potentio-vector)`
```lisp
eustf roseus_c_util hrp2jsknts-init

;; extending gcstack 0x5161e00[32738] --> 0x63ee4e0[65476] top=7f1a
1.irteusgl$ #<hrp2jsknts-robot #X5d823f0 HRP2JSKNTS  0.0 0.0 0.0 / 0.0 0.0 0.0>
2.irteusgl$ send *ri* :state :potentio-vector
/home/jskuser/ros/hydro_parent/devel/share/euslisp/jskeus/eus/Linux64/bin/irteus\
gl 0 error: unbound variable ros::char-vector in (instantiate ros::char-vector r\
os::n)
3.E1-irteusgl$ btrace
euserror
euserror
(instantiate ros::char-vector ros::n)
(setq ros::_driver_temp (instantiate ros::char-vector ros::n))
(let (ros::n) (setq ros::n (system:peek ros::buf ros::ptr- :integer)) (incf ros:\
:ptr- 4) (setq ros::_driver_temp (instantiate ros::char-vector ros::n)) (dotimes\
 (ros::i ros::n) (setf (elt ros::_driver_temp ros::i) (system:peek ros::buf ros:\
:ptr- :short)) (incf ros::ptr- 2)))
(ros::spin-once groupname)
(send self :spin-once)
(let (joint-names positions velocities efforts) (send self :spin-once) (when wai\
t-until-update (send self :wait-until-update-all-joints wait-until-update)) (unl\
ess robot-state (return-from :update-robot-state)) (setq joint-names (cdr (assoc\
 :name robot-state)) positions (cdr (assoc :position robot-state)) velocities (c\
dr (assoc :velocity robot-state)) efforts (cdr (assoc :effort robot-state))) (le\
t ((joint-num (length joint-names))) (when (not (eq joint-num (length velocities\
))) (setq velocities (instantiate float-vector joint-num))) (when (not (eq joint\
-num (length efforts))) (setq efforts (instantiate float-vector joint-num)))) (m\
apcar #'(lambda (n p v e) (let (j (kn (intern (string-upcase n) *keyword-package\
*))) (when (and (find-method robot kn) (setq j (send robot kn))) (send j :ros-jo\
int-angle p) (send j :joint-velocity v) (send j :joint-torque e)))) (coerce join\
t-names cons) (coerce positions cons) (coerce velocities cons) (coerce efforts c\
ons)))
(send self :update-robot-state :wait-until-update (when (position :wait-until-up\
date args) (elt args (+ (position :wait-until-update args) 1))))
t
```

And msg/MotorStates.l has
```lisp
   ;; int16[] _driver_temp                                                                                      
   (let (n)
     (setq n (sys::peek buf ptr- :integer)) (incf ptr- 4)
     (setq _driver_temp (instantiate char-vector n))
     (dotimes (i n)
     (setf (elt _driver_temp i) (sys::peek buf ptr- :short)) (incf ptr- 2)
     ))
```

And we don't have char-vector, is it right?
```
4.E1-irteusgl$ instantiate ros::char-vector 1
/home/jskuser/ros/hydro_parent/devel/share/euslisp/jskeus/eus/Linux64/bin/irteusgl 0 error: unbound variable ro\
s::char-vector in (instantiate ros::char-vector 1)
5.E2-irteusgl$ instantiate char-vector 1
/home/jskuser/ros/hydro_parent/devel/share/euslisp/jskeus/eus/Linux64/bin/irteusgl 0 error: unbound variable ch\
ar-vector in (instantiate char-vector 1)

```

**Comment by YuOhara** [comment]
old ros msgs in .ros/roseus directory has the code like that
```lisp
   ;; int16[] _driver_temp                                                                                                                                                    
   (let (n)
     (setq n (sys::peek buf ptr- :integer)) (incf ptr- 4)
     (setq _driver_temp (instantiate integer-vector n))
     (dotimes (i n)
       (setf (elt _driver_temp i) (sys::peek buf ptr- :short)) (incf ptr- 2)
       ))
``

**Comment by k-okada** [comment]
I see,

it seems both uint8 ,uint16 return `char` for `lisp_type`
https://github.com/jsk-ros-pkg/geneus/blob/master/src/geneus/generate.py#L100

and at
https://github.com/jsk-ros-pkg/geneus/blob/master/src/geneus/generate.py#L133
we use `make-array ..... :element-type :char`

do you think changing retyrn type of uint8/16 to integer works?



--
◉ Kei Okada

**Comment by k-okada** [comment]
ok, fixed in https://github.com/jsk-ros-pkg/geneus/pull/6

**Comment by k-okada** [comment]
this has been tested on https://github.com/jsk-ros-pkg/jsk_roseus/pull/225, but please double check on your applicatoin

**Comment by k-okada** [comment]
I found several bugs (for example if you send minus value with int8, but that is converted to plus value), so it may change the behavior of your software.

**Comment by garaemon** [comment]
Hmm, tough problem, I will check it tomorrow

**Comment by k-okada** [comment]
fixed via #6
